src/string.rs

```diff
@@ -3,8 +3,8 @@ pub fn distance_between_strings_in(
     string2: &str,
     strings: &[String],
 ) -> Option<i32> {
-    let str1_ix = index_of_string_in(string1, &strings);
-    let str2_ix = index_of_string_in(string2, &strings);
+    let str1_ix = index_of_string_in(string1, strings);
+    let str2_ix = index_of_string_in(string2, strings);
```

Comment:
This is the second time I see `AsRef` conversion utilized instead of passing a reference directly, so I am wondering if this is a choice you've made. I tried to find any references to whether this was the idiomatic way to do things, but couldn't find anything concrete. My personal preference is to always pass the reference if the function is expecting a reference, that way:
1. I don't forget to think about how passing owned value might affect the caller of the function if I ever want to go from a reference parameter to an owned one.
2. When I am reading the caller code, it's clear to me that the function will utilize the reference.

Although this is probably a minor point for this project, I'd still like to hear your thoughts on this.

Reply:
To follow up, while reviewing the rest of the PR, I've seen this usage quite a lot. I decided not to comment on it again, so we can discuss all of it here instead.

Reply:
This particular choice was actually suggested by Clippy:

```
warning: this expression borrows a reference (`&[std::string::String]`) that is immediately dereferenced by the compiler
```

Looking at it again, I'm pretty sure this is because `strings` is already a reference, so passing a reference to that reference isn't helpful?

Though I might be misunderstanding your question entirely! 😅

Reply:
OK, so either I didn't realize that the argument was already a reference or I just picked the wrong diff to comment this on since I mentioned that there were several instances. 🤷

I think at this point, it makes sense to merge this in (unless there are any important issues) and discuss issues like this in a new PR when it comes up. I feel both our memories are getting fuzzy with this PR since it's been open for so long.

So, I am going to resolve this one as well.